app/io/pathfinder/controllers/ClusterController.java

```
@@ -0,0 +1,104 @@
+    }
+
+    public Result getCluster(long id) {
+        Cluster cluster = Cluster.find.byId(id);
+
+        if (cluster == null) {
+            return notFound();
+        }
+
+        return ok(jsonContext.toJson(cluster));
+    }
+
+    public Result createCluster() {
+        JsonNode json = request().body().asJson();
+        if (json == null) {
+            return badRequest("Expected content-type text/json or application/json");
+        }
+
+        Cluster cluster;
+
+        try {
+            cluster = Json.fromJson(json, Cluster.class);
+
+            cluster.save();
+            return created(jsonContext.toJson(cluster));
+        } catch (PersistenceException e) {
+            return internalServerError("Error saving cluster to the database: " + e.getMessage());
+        } catch (Exception e) {
+            return badRequest("Unable to map json to commodity object: " + e.getMessage());
+        }
+    }
+
+    public Result editCluster(long id) {
+        Cluster cluster = Cluster.find.byId(id);
+
+        if (cluster == null) {
+            return notFound();
+        }
+
+        ObjectNode clusterJson = (ObjectNode) Json.toJson(cluster);
+        ObjectNode body;
+
+        try {
+            JsonNode json = request().body().asJson();
+            body = (ObjectNode) json;
+        } catch (ClassCastException e) {
+            return badRequest("Cannot cast request body to ObjectNode: " + e.getMessage());
+        }
+
+        Iterator<String> fields = clusterJson.fieldNames();
+        while (fields.hasNext()) {
+            String field = fields.next();
+
+            if (field.equals("id")) {
+                continue;
+            }
+
+            if (body.has(field)) {
+                clusterJson.replace(field, body.findPath(field));
+            }
+        }
+
+        try {
+            cluster = Json.fromJson(clusterJson, Cluster.class);
+
+            cluster.update();
+
+            return noContent();
+        } catch (Exception e) {
```

Can we catch a more granular exception here? I don't think we should every catch all exceptions.

For example, if we catch a java.lang.InterruptedException it fucks with the JVM's scheduler.